dlt/sources/helpers/rest_client/auth.py

```diff
@@ -138,6 +139,72 @@ def __call__(self, request: PreparedRequest) -> PreparedRequest:
         return request
 
 
+@configspec
+class OAuth2ImplicitFlow(OAuth2AuthBase):
+    """
+    This class implements OAuth2 implicit flow which does not require the end user to
+    give permission to the app to access their information. This is often used for
+    server-to-server authentication.
+    The client obtains a temporary access token from the authorization service.
+    With the access token, the client can access resource services
+    """
+
+    def __init__(
+        self,
+        access_token_url: str,
+        access_token_request_data: Dict[str, Any],
+        client_id: TSecretStrValue,
+        client_secret: TSecretStrValue,
+        access_token: TSecretStrValue = None,
+        default_token_expiration: int = 3600,
+    ) -> None:
+        super().__init__(access_token)
+        self.access_token_request_data = access_token_request_data
+        self.access_token_url = access_token_url
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.default_token_expiration = default_token_expiration
+        self.token_expiry: pendulum.DateTime = pendulum.now()
+
+    def __call__(self, request: PreparedRequest) -> PreparedRequest:
+        if self.access_token is None or self.is_token_expired():
+            self.obtain_token()
+        request.headers["Authorization"] = f"Bearer {self.access_token}"
+        return request
+
+    def is_token_expired(self) -> bool:
+        return pendulum.now() >= self.token_expiry
+
+    @abstractmethod
+    def build_access_token_request(self) -> Dict[str, Any]:
+        pass
+
+    def obtain_token(self) -> None:
+        response = requests.post(**self.build_access_token_request())
+        response.raise_for_status()
+        self.access_token = response.json()["access_token"]
+        expires_in = response.json().get("expires_in", self.default_token_expiration)
+        self.token_expiry = pendulum.now().add(seconds=expires_in)
+
+
+class OAuth2Zoom(OAuth2ImplicitFlow):
```

Comment:
Hey @willi-mueller I think once this out of the draft state this class would be too specific for this file. We could keep the code nonetheless and I would suggest moving this to the docs as an example of OAuth. (Right now there's nothing in the docs related to handling OAuth). What do you think?